homeassistant/components/zha/binary_sensor.py

```diff
@@ -60,258 +70,103 @@
-        state = result.get('zone_status', self._state)
-        if isinstance(state, (int, uint16_t)):
-            self._state = result.get('zone_status', self._state) & 3
-
-
-class Remote(RestoreEntity, ZhaEntity, BinarySensorDevice):
-    """ZHA switch/remote controller/button."""
-
-    _domain = DOMAIN
-
-    def __init__(self, **kwargs):
-        """Initialize Switch."""
-        super().__init__(**kwargs)
-        self._level = 0
-        from zigpy.zcl.clusters import general
-        self._out_listeners = {
-            general.OnOff.cluster_id: OnOffListener(
-                self,
-                self._out_clusters[general.OnOff.cluster_id]
-            )
-        }
-
-        out_clusters = kwargs.get('out_clusters')
-        self._zcl_reporting = {}
-
-        if general.LevelControl.cluster_id in out_clusters:
-            self._out_listeners.update({
-                general.LevelControl.cluster_id: LevelListener(
-                    self,
-                    out_clusters[general.LevelControl.cluster_id]
-                )
-            })
+        if self._level_listener:
+            await self.async_accept_signal(
+                self._level_listener, SIGNAL_SET_LEVEL, self.set_level)
+            await self.async_accept_signal(
+                self._level_listener, SIGNAL_MOVE_LEVEL, self.move_level)
+        if self._on_off_listener:
+            await self.async_accept_signal(
+                self._on_off_listener, SIGNAL_ATTR_UPDATED, self.set_state)
+        if self._zone_listener:
+            await self.async_accept_signal(
+                self._zone_listener, SIGNAL_ATTR_UPDATED, self.set_state)
+        if self._attr_listener:
+            await self.async_accept_signal(
+                self._attr_listener, SIGNAL_ATTR_UPDATED, self.set_state)
 
     @property
     def is_on(self) -> bool:
-        """Return true if the binary sensor is on."""
+        """Return if the switch is on based on the statemachine."""
+        if self._state is None:
+            return False
         return self._state
 
     @property
-    def device_state_attributes(self):
-        """Return the device state attributes."""
-        self._device_state_attributes.update({
-            'level': self._state and self._level or 0
-        })
-        return self._device_state_attributes
+    def device_class(self) -> str:
+        """Return device class from component DEVICE_CLASSES."""
+        return self._device_class
 
-    @property
-    def zcl_reporting_config(self):
-        """Return ZCL attribute reporting configuration."""
-        return self._zcl_reporting
+    def set_state(self, state):
+        """Set the state."""
+        self._state = bool(state)
+        self.async_schedule_update_ha_state()
 
     def move_level(self, change):
         """Increment the level, setting state if appropriate."""
+        level = self._temporary_state_attributes.get(LEVEL, 0)
```

Comment:
`_temporary_state_attributes` needs clarification why it's needed in a docstring or comment.

Reply:
I may have a way to remove this. Will explore this evening

Reply:
gone